### PR TITLE
feat[w3.4]: observe patient's appointments

### PR DESCRIPTION
**Pull request by sladjana-savkovic**
The goals of this PR are:
-  Update backend code (refactoring, adding new methods, handling exceptions improvement)
-  Modify factory pattern for creating test objects
-  Make unit tests
-  Add new methods in controller
-  Make frontend
-  Connect frontend and backend

### CODE REVIEW

**Comment by sladjana-savkovic** [comment]
[(5) W3.4 : As a patient I want to observe my appointments, so that I could check my appointment's schedule.](https://trello.com/c/zDvaBaVg/31-5-w34-as-a-patient-i-want-to-observe-my-appointments-so-that-i-could-check-my-appointments-schedule)

**Comment by milana-todorovic** [comment]
I renamed your pull request so it's clearer what functionality it refers to.

File: HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs [hunk 1]
```diff
@@ -65,5 +65,76 @@ public ActionResult AdvanceSearchExaminations(ExaminationSearchDTO examinationSe
...
         }
 
+        /// <summary>
+        /// /getting canceled examinations linked to a certain patient
+        /// </summary>
+        /// <param name="patientJmbg">patients jmbg</param>
+        /// <returns>if alright returns code 200(Ok), if patuentJmbg is null returns 400, if connection lost returns 500</returns>
+        [HttpGet("{canceled/patientJmbg}")]
+        public ActionResult GetCanceledExaminationsByPatient(string patientJmbg)
+        {
```
Comment (milana-todorovic) [inline HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs]: Shouldn't it be "cancelled/{patientJmbg}"? I think you accidentally put the bracket in the wrong place.

File: HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs [hunk 2]
```diff
@@ -65,5 +65,76 @@ public ActionResult AdvanceSearchExaminations(ExaminationSearchDTO examinationSe
...
+        }
+
+        /// <summary>
+        /// /getting following examinations linked to a certain patient
+        /// </summary>
+        /// <param name="patientJmbg">patients jmbg</param>
+        /// <returns>if alright returns code 200(Ok), if patuentJmbg is null returns 400, if connection lost returns 500</returns>
+        [HttpGet("{previous/patientJmbg}")]
+        public ActionResult GetFollowingExaminationsByPatient(string patientJmbg)
+        {
```
Comment (milana-todorovic) [inline HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs]: You accidentally left the same path here as in the method for previous examinations, I assume you intended something like "following/{patientJmbg}".

**Review by milana-todorovic — COMMENTED** [review]
I really like the improvements you made to the error handling here! And I also like the decision to remove the overly restrictive interface from the classes for test object creation, I think they'll be much easier to use this way.
Also, some changes were applied both on this branch and on PR #99 (the fields that were added to Examination), but I don't think it should cause issues. I assume that that #99 will merge first though, since you also plan to do the frontend on this branch?

File: HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs (hunk 1 above)
Comment (sladjana-savkovic) [reply HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs]: Yes. I made a mistake at one place, and just copied it. I have plan to do frontend on this branch, so PR #99 will merge earlier.

Comment (milana-todorovic) [reply HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs]: Fixed in af1b49d1baa111f3b0ee80cf8348c9c52a4adcae.

File: HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs (hunk 2 above)
Comment (milana-todorovic) [reply HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs]: Fixed in af1b49d1baa111f3b0ee80cf8348c9c52a4adcae.

**Review by anciz — DISMISSED** [review]

File: HealthcareSystem/PatientWebApp/wwwroot/html/patient_registration.html [hunk 3]
```diff
@@ -17,7 +17,7 @@
     <link rel="stylesheet" href="../css/style.css">
     <link rel="stylesheet" href="../css/style_for_patient_registration.css">
     <script src="../lib/jquery/jquery.min.js"></script>
-    <script type="text/javascript" src="../js/patient_registration.js"></script>
+    <script type="text/javascript" src="../js/patient_examinations.js"></script>
```
Comment (milana-todorovic) [inline HealthcareSystem/PatientWebApp/wwwroot/html/patient_registration.html]: Is this a mistake? I don't think the registration page has a reason to be linked to the JavaScript for examinations?

**Review by milana-todorovic — APPROVED** [review]
Seems alright!

File: HealthcareSystem/PatientWebApp/wwwroot/html/patient_registration.html (hunk 3 above)
Comment (milana-todorovic) [reply HealthcareSystem/PatientWebApp/wwwroot/html/patient_registration.html]: Fixed in f8e2cdf43674f6ce7ea5ae62290ce8a8bd517e4a.

**Review by anciz — APPROVED** [review]
Seems to be right.

**Pull request merged**